zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/immutable/DeployableResource.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.immutable;
+
+import io.camunda.zeebe.protocol.impl.record.value.deployment.DeploymentRecord;
+import org.agrona.DirectBuffer;
+
+/** A record that contain a resource that can be added to a {@link DeploymentRecord */
```

Comment:
Note that:
- `DeployableResource` and all of its implementation are part of `engine` 
- `DeploymentRecord` is part of `protocol-impl`
I would have preferred the other way around but looks like all records in `protocol-impl`  don't have many methods so I figured this is better. Happy to change it if needed